pumpkin/src/world/mod.rs

```diff
@@ -203,6 +204,68 @@ impl World {
         player.send_permission_lvl_update().await;
         client_cmd_suggestions::send_c_commands_packet(&player, command_dispatcher).await;
 
+        if ADVANCED_CONFIG.server_links.enabled {
+            let mut links: Vec<Link> = Vec::new();
+
+            let bug_report = &ADVANCED_CONFIG.server_links.bug_report;
+            if !bug_report.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::BugReport), bug_report))
+            }
+
+            let support = &ADVANCED_CONFIG.server_links.support;
+            if !support.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Support), support))
+            }
+
+            let status = &ADVANCED_CONFIG.server_links.status;
+            if !status.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Status), status))
+            }
+
+            let feedback = &ADVANCED_CONFIG.server_links.feedback;
+            if !feedback.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Feedback), feedback))
+            }
+
+            let community = &ADVANCED_CONFIG.server_links.community;
+            if !community.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Community), community))
+            }
+
+            let website = &ADVANCED_CONFIG.server_links.website;
+            if !website.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Website), website))
+            }
+
+            let forums = &ADVANCED_CONFIG.server_links.forums;
+            if !forums.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Forums), forums))
+            }
+
+            let news = &ADVANCED_CONFIG.server_links.news;
+            if !news.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::News), news))
+            }
+
+            let announcements = &ADVANCED_CONFIG.server_links.announcements;
+            if !announcements.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Announcements), announcements))
+            }
+
+            for (key, value) in &ADVANCED_CONFIG.server_links.custom {
+                // add
```

Comment:
Got some lifetime and borrowing issue here, would be nice if a rust pro could explain it to me 👉👈

Reply:
```rs
for (key, value) in &ADVANCED_CONFIG.server_links.custom {
    links.push(Link::new(&Label::TextComponent(TextComponent::text(key)), value))
}
```

```
error[E0716]: temporary value dropped while borrowed                                                                                                                                                        
   --> pumpkin\src\world\mod.rs:297:39
    |
297 |                 links.push(Link::new(&Label::TextComponent(TextComponent::text(key)), value))
    |                 -----                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
    |                 |
    |                 borrow later used here
298 |             }
    |             - temporary value is freed at the end of this statement
    |
    = note: consider using a `let` binding to create a longer lived value

For more information about this error, try `rustc --explain E0716`.
error: could not compile `pumpkin` (bin "pumpkin") due to 1 previous error
```